### ISSUE TITLE
Umsetzung Monopsen in Fassungsansicht

### ISSUE THREAD

**Issue by flicksolutions** [DHBern/presentation_parzival#17]
* link an all-encompassing Monopsen PDF file
  * perhaps link to specific page in the PDF from more specific linking locations (TBD)
  * the PDF also contains lnks to the representation in the edition; it will need to be regenerated to point to the new edition

**Comment by flicksolutions** [comment]
We will link to the same PDF that contains the complete text - with a varying link to the Dreissiger that is active.

**Comment by pdaengeli** [comment]
@flicksolutions, I got a notification that a number of `mon` PDFs were uploaded to Dropbox today, seemingly one per Dreissiger.

Do we need these? Or rather an all-encompassing PDF? 

In either case, should we offer the file(s) through the static API?

**Comment by pdaengeli** [comment]
Merging the single files using `pdfunite $(ls -v *.pdf) monopsen.pdf` yields [monopsen.pdf](https://github.com/user-attachments/files/18524266/monopsen.pdf).

There is slight variation in the dimension/width of the pages.

Equal size is possible using TeX / `pdfjam --outfile monopsen-a4.pdf --paper a4paper $(ls -v *.pdf)`: [monopsen-a4.zip](https://github.com/user-attachments/files/18524538/monopsen-a4.zip)

As far as I can tell the page number equals the Dreissiger number in both cases.

Should I include one of these in the static API?

**Comment by flicksolutions** [comment]
hmm, the links are in a old format anyways (using Query-Attributes instead of a path).
I have to think about if I get make these links work in the new page aswell in a simple way...

If I can make it work I would prefer monopfen-pdftk.pdf because of the retained links.
I keep you posted.

**Comment by pdaengeli** [comment]
From the OP:

> * the PDF also contains lnks to the representation in the edition; it will need to be regenerated to point to the new edition

Having SA regenerating the PDFs is the most straightforward approach.

Otherwise we'd need to set up URL rewriting within the web app (perhaps SvelteKit has some kind of adapter to this end) or (ab)use GH Pages 404 pages to do redirects (this is possible with a mapping file [like so](https://github.com/pdaengeli/404/) but kind of hacky) or use an actual webserver to proxy the requests.

**Comment by flicksolutions** [comment]
I agree. Rewriting with SvelteKit is possible. But if SA can help us, it would surely be preferable.
I'll contact him and ask.

**Comment by pdaengeli** [comment]
@flicksolutions, no need. I reached out yesterday.

**Comment by flicksolutions** [comment]
Perfect, thanks! Schema would be like so: https://www.parzival.unibe.ch/fassungen/[dreissiger]

**Comment by pdaengeli** [comment]
FWIW, it isn't hard to replace the links:

```
pdftk monopsen-pdftk.pdf output foo.tmp
parallel sed -i -e 's/parzdb.*dreissiger=\([0-9]\+\).*\()\)/fassungen\/\1\2/' -- foo.tmp
pdftk foo.tmp output monopsen-pdftk-updated-links.pdf compress
```

[monopsen-pdftk-updated-links.pdf](https://github.com/user-attachments/files/20090111/monopsen-pdftk-updated-links.pdf)

Using `pdftk` we get pages of varying size, but it might be good enough.

**Comment by flicksolutions** [comment]
Very good to know for the 21.5.! Thank you!
